UPGRADE-6.0.md

```diff
@@ -0,0 +1,131 @@
+- `snapshot_copy` block in `aws_redshift_cluster`
+
+### Variable and output changes
+
+1. Removed variables:
+
+  - Cluster
+    - `var.logging.enable` has been removed
+
+2. Renamed variables:
+
+  - None
+
+3. Added variables:
+
+  - Snapshot Copy
+    - `var.snapshot_copy.manual_snapshot_retention_period`
+
+4. Removed outputs:
+
+  - None
+
+5. Renamed outputs:
+
+  - None
+
+6. Added outputs:
+
+  - None
+
+## Upgrade Migration
+
+### Before v5.x Example
+
+```hcl
+module "redshift" {
+  source  = "terraform-aws-modules/redshift/aws"
+  version = "5.4.0"
+
+  snapshot_copy = {
+    destination_region = "us-east-1"
+    grant_name         = aws_redshift_snapshot_copy_grant.useast1.snapshot_copy_grant_name
+  }
+
+  logging = {
+    enable        = true
+    bucket_name   = module.s3_logs.s3_bucket_id
+    s3_key_prefix = local.s3_prefix
+  }
+}
+```
+
+### After v6.x Example
+
+```hcl
+module "redshift" {
+  source  = "terraform-aws-modules/redshift/aws"
+  version = "6.0.0"
+
+  snapshot_copy = {
+    destination_region = "us-east-1"
+    grant_name         = aws_redshift_snapshot_copy_grant.useast1.snapshot_copy_grant_name
+  }
+
+  logging = {
+    bucket_name   = module.s3_logs.s3_bucket_id
+    s3_key_prefix = local.s3_prefix
+  }
+}
+```
+
+### Diff of Before vs After
+
+```diff
+  # module.redshift.aws_redshift_logging.this[0] will be created
+  + resource "aws_redshift_logging" "this" {
+      + bucket_name        = "ex-complete20240414012816938100000003"
+      + cluster_identifier = "ex-complete"
+      + id                 = (known after apply)
+      + s3_key_prefix      = "redshift/ex-complete/"
+    }
+
+  # module.redshift.aws_redshift_snapshot_copy.this[0] will be created
+  + resource "aws_redshift_snapshot_copy" "this" {
+      + cluster_identifier               = "ex-complete"
+      + destination_region               = "us-east-1"
+      + id                               = (known after apply)
+      + manual_snapshot_retention_period = (known after apply)
+      + retention_period                 = (known after apply)
+      + snapshot_copy_grant_name         = "ex-complete-us-east-1"
+    }
+```
+The `aws_redshift_logging` can be applied or imported. If setting the `log_destination_type`, an apply following an import will be required to clear the remaining diff.
+The `aws_redshift_snapshot_copy` resource requires importing if an existing snapshot_copy configuration exists.
+
+### State Move Commands
+
+None required
+
+### State Import Commands
+
+During the migration to v6.x of this module, logging and snapshot_copy resources will be created by this module if those settings are configured. In order to guarantee the best experience and prevent data loss, you will need to import them into terraform state using commands like these:
+
+```bash
+terraform import 'module.redshift.aws_redshift_logging.this[0]' <cluster-id>
+terraform import 'module.redshift.aws_redshift_snapshot_copy.this[0]' <cluster-id>
```

Comment:
For those who have `snapshot_copy` previously configured, this will require an import or it will produce an error that snapshot copy is already enabled when it attempts to create the new resource. I don't mind removing this guide but just want to confirm that we don't want to consider this a breaking change.

```
╷
│ Error: creating Amazon Redshift Snapshot Copy ("ex-complete"): operation error Redshift: EnableSnapshotCopy, https response error StatusCode: 400, RequestID: caf48715-ac18-4a26-9b28-5c8b647cc5e7, SnapshotCopyAlreadyEnabledFault: Snapshot Copy is already enabled on Cluster ex-complete
│ 
│   with module.redshift.aws_redshift_snapshot_copy.this[0],
│   on ../../main.tf line 321, in resource "aws_redshift_snapshot_copy" "this":
│  321: resource "aws_redshift_snapshot_copy" "this" {
│ 
│ operation error Redshift: EnableSnapshotCopy, https response error StatusCode: 400, RequestID: caf48715-ac18-4a26-9b28-5c8b647cc5e7, SnapshotCopyAlreadyEnabledFault: Snapshot Copy is already enabled on Cluster ex-complete
╵
```

Reply:
I see. I thought these resources had the same adoptable behavior as S3 bucket resources like "logging," "versioning," etc. Thank you for the explanation.